patches/0007-Add-Darwin-crypto-backend.patch

```diff
@@ -0,0 +1,601 @@
+ src/crypto/internal/backend/darwin_darwin.go  | 337 ++++++++++++++++++
+ src/crypto/internal/backend/fips140/darwin.go |  11 +
+ src/crypto/rsa/boring.go                      |   1 -
+ src/crypto/rsa/darwin.go                      |  71 ++++
+ src/go/build/vendor_test.go                   |   1 +
+ src/net/lookup_test.go                        |   3 +
+ src/runtime/pprof/vminfo_darwin_test.go       |   6 +
+ 10 files changed, 475 insertions(+), 1 deletions(-)
+ create mode 100644 src/crypto/internal/backend/bbig/big_darwin.go
+ create mode 100644 src/crypto/internal/backend/darwin_darwin.go
+ create mode 100644 src/crypto/internal/backend/fips140/darwin.go
+ create mode 100644 src/crypto/rsa/darwin.go
+
+diff --git a/.gitignore b/.gitignore
+index c6512e64a4ef39..b3b01db73b009d 100644
+--- a/.gitignore
++++ b/.gitignore
+@@ -46,6 +46,8 @@ _testmain.go
+ /test/run.out
+ /test/times.out
+ 
++!/src/vendor/github.com/microsoft/go-crypto-darwin/internal/cryptokit/CryptoKit.o
++
+ # This file includes artifacts of Go build that should not be checked in.
+ # For files created by specific development environment (e.g. editor),
+ # use alternative ways to exclude files from git.
+diff --git a/src/crypto/internal/backend/bbig/big_darwin.go b/src/crypto/internal/backend/bbig/big_darwin.go
+new file mode 100644
+index 00000000000000..77f3ca5d262769
+--- /dev/null
++++ b/src/crypto/internal/backend/bbig/big_darwin.go
+@@ -0,0 +1,12 @@
++// Copyright 2022 The Go Authors. All rights reserved.
++// Use of this source code is governed by a BSD-style
++// license that can be found in the LICENSE file.
++
++//go:build goexperiment.darwincrypto && cgo
++
++package bbig
++
++import "github.com/microsoft/go-crypto-darwin/bbig"
++
++var Enc = bbig.Enc
++var Dec = bbig.Dec
+diff --git a/src/crypto/internal/backend/common.go b/src/crypto/internal/backend/common.go
+index 84447174284ffd..f85f3b6c865764 100644
+--- a/src/crypto/internal/backend/common.go
++++ b/src/crypto/internal/backend/common.go
+@@ -5,8 +5,10 @@
+ package backend
+ 
+ import (
++	"crypto"
+ 	"crypto/internal/backend/fips140"
+ 	"crypto/internal/boring/sig"
++	"internal/goexperiment"
+ 	"runtime"
+ )
+ 
+@@ -56,3 +58,33 @@ func UnreachableExceptTests() {
+ 		}
+ 	}
+ }
++
++func IsCurveSupported(curve string) bool {
++	switch curve {
++	case "P-256", "P-384", "P-521":
++		return true
++	case "P-224":
++		return !goexperiment.DarwinCrypto
++	}
++	return false
++}
++
++func IsRSAOAEPLabelSupported(label []byte) bool {
++	if goexperiment.DarwinCrypto {
++		// CommonCrypto doesn't support labels
++		// https://github.com/microsoft/go-crypto-darwin/issues/22
++		return len(label) == 0
++	}
++	return true
++}
++
++func IsPKCS1v15HashSupported(hash crypto.Hash) bool {
++	if goexperiment.DarwinCrypto {
++		switch hash {
++		case crypto.SHA1, crypto.SHA224, crypto.SHA256, crypto.SHA384, crypto.SHA512, 0:
++			return true
++		}
++		return false
++	}
++	return true
++}
```

Comment:
Move this functions to the backend files (as we do for other Supports* helpers). Also, rename the functions to follow the  `Supports*` pattern instead of the `Is*Supported` pattern.

Reply:
updated PTAL